sql-common/client.cc

```diff
@@ -4427,14 +4415,19 @@ static net_async_status cli_establish_ssl_nonblocking(MYSQL *mysql, int *res) {
     /* sslconnect creates a new vio, so update it. */
     vio_set_blocking_flag(net->vio, !ctx->non_blocking);
 
+    /* OpenSSL 1.0.2 and up will use their own verification function.
+     * This is set up in new_VioSSLConnectorFd and adding here it here
+     * again is superfluous.
+     */
+#if OPENSSL_VERSION_NUMBER < 0x10002000L
```

Comment:
On OpenSSL 1.0.2 and newer, the additional identity check here is not needed since we already set it up inside `new_VioSSLConnectorFd` using the OpenSSL provided validation function. This means the check here was superfluous and would result in double checking the certificate.

If anything, it makes it more brittle because there could be implementation differences between these validation functions (such as honoring DNS SANs, using the deprecated CN or not etc).